modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/mv/PageMemoryMvPartitionStorage.java

```diff
@@ -338,24 +353,38 @@ private VersionChain findVersionChainForModification(RowId rowId) {
             throw new NoUncommittedVersionException();
         }
 
-        RowVersion currentVersion = findLatestRowVersion(currentVersionChain, ALWAYS_LOAD_VALUE);
-        assert currentVersion.isUncommitted();
+        RowVersion latestVersion = findLatestRowVersion(currentVersionChain, ALWAYS_LOAD_VALUE);
+        assert latestVersion.isUncommitted();
+
+        removeRowVersion(latestVersion);
 
-        removeRowVersion(currentVersion);
+        if (latestVersion.hasNextLink()) {
+            // This load can be avoided, see the comment below.
+            RowVersion preLatestVersion = readPredecessor(latestVersion);
```

Comment:
I think the problem is caused by the fact that we have 2 directions. One is chronological (previous is what was created before, next is what was created later), another one is 'chain order' ('next' in the chain is what was created before). This is because elements in our linked list come backwards.

The 'order of chain' is important on a lower level, and on the higher level it seems to be more convenient  to use the 'creation order', so I use 'latest' as the term to mean 'the version created most recently'.

The problem is when we go from one terminology ('next' in the 'chain order') to another ('predecessor' in the 'creation order').

I see the following possibilities:

1. Change the terminology on the chain level: 'tail' instead of 'head', 'previous' instead of 'next'. Radical and contradicts with the usual terminology.
2. Keep head/next terminology on the chain level, but on the store level, avoid using 'latest', instead use 'current'. But it's unclear how to deal with the version that was created just before the 'current'. Pre-current or next-to-current?
3. Have a convention that we separate the levels. On the lower (chain) level, we use head/next terminology; on the higher (store) level, we use latest/predecessor terminology. Can still be confusing. This is what I implemented for now as a response to your comment.

WDYT?

Reply:
Renamed the variable to `latestCommittedVersion`